File: apps/servers/app/schemas/recipe.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

# Shared properties
class RecipeBase(BaseModel):
    title: str
    description: Optional[str] = None
    ingredients: Optional[str] = None
    instructions: Optional[str] = None
    image_url: Optional[str] = None


# Schema for creating a recipe
class RecipeCreate(RecipeBase):
    pass


# Schema for reading a recipe (from DB)
class RecipeRead(RecipeBase):
    id: int
    owner_id: int
    created_at: datetime
    updated_at: datetime

    class Config:
        orm_mode = True

File: apps/web/src/types/recipe.ts
export interface Recipe {
  id: number;
  title: string;
  description?: string;
  ingredients: string;
  steps: string;
  tags?: string;
  image?: string;
  created_by?: number;
  created_at: string;
}

export interface RecipeCreate {
  title: string;
  description?: string;
  ingredients: string;
  steps: string;
  tags?: string;
  image?: string;
}
